### PR TITLE
Force UTF-8 in JSON responses

### PR DESCRIPTION
**Pull request by Taikelenn**
# Description of the PR

By [RFC 8259](https://datatracker.ietf.org/doc/html/rfc8259) section 8.1, JSON must be encoded as UTF-8. However this doesn't occur automatically for some reason (perhaps due to non-standard Content-Type header in A+ responses), so we enforce the proper encoding.

In practice, this used to be an issue only on Windows because AFAIK modern Linux distros use UTF-8 everywhere. Windows, however, uses language-specific code pages :(

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [X] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [X] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [X] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [X] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/model/Course.java [hunk 1]
```diff
@@ -190,7 +191,7 @@ public static Course fromUrl(@NotNull URL url, @NotNull ModelFactory modelFactor
       throws IOException, MalformedCourseConfigurationException {
     InputStream inputStream = CoursesClient.fetch(url);
     return Course.fromConfigurationData(
-        new InputStreamReader(inputStream), url.toString(), modelFactory);
+        new InputStreamReader(inputStream, StandardCharsets.UTF_8), url.toString(), modelFactory);
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/model/Course.java:194]: Hmm... 🤔 Ideally, this kind of manipulation would happen much closer to the DAL but it's not a problem of this PR, _per se_.

**Review by OlliKiljunen — APPROVED** [review]
Very good! 👍

File: src/main/java/fi/aalto/cs/apluscourses/model/Course.java (hunk 1 above)
Comment (Taikelenn) [reply src/main/java/fi/aalto/cs/apluscourses/model/Course.java:194]: I guess ideally we should have a method CoursesClient.fetchJSON which would be used for reading all JSONs and this method would already handle UTF-8

**Review by jaakkonarhi — APPROVED** [review]
nice

**Pull request merged**